api-conversion/src/main/java/io/strimzi/kafka/api/conversion/cli/ConvertResourceCommand.java

```diff
@@ -0,0 +1,254 @@
+    private CustomResource getNamedResource(String kind, String name, String namespace)    {
+        MixedOperation<CustomResource, CustomResourceList, ?> op = DEFAULT_OPERATIONS.get(kind).apply(client);
+
+        return op.inNamespace(namespace).withName(name).get();
+    }
+
+    /**
+     * Gets resources of given Kinds from one or all namespaces
+     *
+     * @param kinds             Array with Kinds which should be converted
+     * @param namespace         The namespace in which the resources should be converted
+     * @param allNamespaces     Indicates to convert resources in all namespaces
+     *
+     * @return                  List of found resources which should be converted
+     */
+    private List<CustomResource> getResources(String[] kinds, String namespace, boolean allNamespaces)    {
+        List<CustomResource> crs = new ArrayList<>();
+
+        for (String kind : kinds)   {
+            crs.addAll(get(kind, namespace, allNamespaces));
+        }
+
+        return crs;
+    }
+
+    /**
+     * Converts the resource to the new APi version
+     *
+     * @param cr    Custom resource which should be converted
+     *
+     * @return      The converted custom resource
+     */
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    protected CustomResource convert(CustomResource cr) {
+        log.info("Converting {} resource named {} from namespace {}", cr.getKind(), cr.getMetadata().getName(), cr.getMetadata().getNamespace());
+        getConverter(cr.getClass()).convertTo(cr, TO_API_VERSION);
+        return cr;
+    }
+
+    /**
+     * When multi-part conversions occurred (conversion which resulted in multiple YAMLs), we have to create them using
+     * the Kubernetes client.
+     *
+     * @param namespace     Namespace where the resource should be created
+     */
+    private void handleMultipartResources(String namespace) {
+        try {
+            List<MultipartResource> resources = MultipartConversions.get().getResources();
+            for (MultipartResource resource : resources) {
+                resource.getK8sConsumer().accept(client, namespace);
+            }
+        } finally {
+            MultipartConversions.remove();
+        }
+    }
+
+    /**
+     * Converts the Custom Resource in Kubernetes
+     *
+     * @param cr            Custom resource to be converted
+     *
+     * @throws IOException  Exception if the replacement fails
+     */
+    private void convertInKube(CustomResource cr) throws IOException {
+        CustomResource convertedCr = convert(cr);
+        handleMultipartResources(cr.getMetadata().getNamespace());
+        replace(cr.getKind(), convertedCr);
+    }
+
+    /**
+     * Reads the resources from the Kubernetes API and converts them
+     */
+    @Override
+    public void run() {
+        try {
+            String namespace;
+            boolean allNamespaces;
+            client = new DefaultKubernetesClient();
+            Crds.registerCustomKinds();
```

Comment:
Probably better to do this in a static initializer, so they're definitely only registered once.